### ISSUE TITLE
Solid block characters are invisible under Windows Vista

### ISSUE THREAD

**Issue by angband-import** [angband/angband#774]
**Reported by ctate on 9 Jul 2007 06:14 UTC**
The solid block character generally referred to with character code 127 displays as a black space under Vista.  This makes Angband unplayable as supplied in the standard .zip Windows distribution, since that invisible character is configured as the wall character.

Ideally the code should be fixed to use an appropriate display mechanism for graphics characters under Vista.  A workaround might be to use the default '#' character for walls in the standard distribution.

**Comment by angband-import** [comment]
**Comment by takkaria on 9 Jul 2007 10:04 UTC**
See the FAQ.  You just have to delete the block character lines.

We'll move to the SDL port eventually, I imagine, and this problem will go then.

**Comment by angband-import** [comment]
**Comment by anonymous on 31 Jul 2007 03:02 UTC**
Thanks for nothing.

**Comment by angband-import** [comment]
**Comment by takkaria on 31 Jul 2007 11:54 UTC**
I'm sorry, but either we remove all the pseudographics and get lots of complaints about needing them, and everyone who wants them has to fiddle with pref files to put them in, or Vista users, of whom there aren't currently that many, have to take out the lines.  Given that the latter inconveniences fewer people than the former, that's the only reasonable decision to make.

**Comment by angband-import** [comment]
**Comment by jasonite10\@aol.com on 12 Apr 2008 08:56 UTC**
I'm a Vista user (as well as an XP user) and I finally installed a few Angband variants on my machine.... what a surprise this was.

I hope a solution can be found soon, it's too difficult to play without 'em!  :)

**Comment by angband-import** [comment]
**Comment by takkaria on 30 Apr 2008 00:09 UTC**
Reopening this since it's something we want to fix.  The very least we could do would be to detect whether Vista is being used or XP and switch to a different set of pref files if so.

(Though I'd still rather like to switch to SDL long-term.)

**Comment by angband-import** [comment]
**Modified by takkaria on 28 Jun 2008 10:02 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 7 Jan 2009 08:23 UTC**
[712d286] (SVN r1076) fixes this, but I'm reassinging to 3.1.1 because this needs a proper fix soon.  The proper fix would involve going through all the font files that Nick hasn't and moving the wall character from its current position from 127 to something below character 16.

Nick's advice is as follows:
```
1. Sometimes (maybe always?) Windows doesn't like using characters
   16-31, or character 127 in the font. *Maintainers should avoid
   these in font-win.prf.*
2. Vista insists on using it's own 8x13 font, which has none of the
   useful things like wall blocks in it. *Players using Vista should
   not use 8x13 as a font size, and maintainers should use a
   different default size (say 18x12) in main-win.c.*

The other issue is that the default angband fonts were 75x75 dpi, and
Vista apparently requires 96x96.  I am pretty sure that all the fonts
on my page have been tested in Vista.  Also, the font-win.prf in FA0.3.5a
should be a good one to check against (except of course that some terrain
values may be different).
```

**Comment by angband-import** [comment]
**Comment by takkaria on 8 Jan 2009 09:30 UTC**
http://angband.oook.cz/faangband/Fonts.html is the page with the existing converted fonts on it.

**Comment by angband-import** [comment]
**Modified by magnate on 11 Jul 2009 20:26 UTC**

**Comment by angband-import** [comment]
**Modified by takkaria on 25 Jul 2009 16:53 UTC**

**Comment by angband-import** [comment]
**Comment by takkaria on 25 Jul 2009 17:47 UTC**
bumping this fix to 3.1.2... again

**Comment by angband-import** [comment]
**Comment by magnate on 21 Nov 2010 20:02 UTC**
Punting to 3.3: non-urgent bug or change.

**Comment by angband-import** [comment]
**Comment by takkaria on 23 Dec 2010 04:50 UTC**
fixed in 3.2 already w Nick's xchars work
